### PR TITLE
[OSD-6575] Enrich documentation

### PR DESCRIPTION
**Pull request by dastergon**
* Added a section describing the reuse/cleanup workflow and added diagram illustrating it
* Improved clarity in some areas of the documentation 
*  Improved the markdown formatting

PTAL

### CODE REVIEW

File: docs/3.0-Controllers.md [hunk 1]
```diff
@@ -198,7 +219,7 @@ awsCredsAccessKeyID     = "aws_access_key_id"
 awsCredsSecretAccessKey = "aws_secret_access_key"
 ```
 
-### 3.3.2. Spec
+### 3.3.3. Spec
 
 Updates the accountClaim CR
```
Comment (dkeohane) [inline docs/3.0-Controllers.md:224]: `AccountClaim`

File: docs/3.0-Controllers.md [hunk 2]
```diff
@@ -240,25 +261,25 @@ status:
     state: Ready
 ```
 
-*state* can be any of the ClaimStatus strings defined in accountclaim_types.go
-*conditions* indicates the last state the account had and supporting details
+* `state` can be any of the ClaimStatus strings defined in accountclaim_types.go
```
Comment (dkeohane) [inline docs/3.0-Controllers.md:264]: Link to file `accountclaim_types.go` [https://github.com/openshift/aws-account-operator/blob/master/pkg/apis/aws/v1alpha1/accountclaim_types.go#L84](url)

File: docs/3.0-Controllers.md [hunk 3]
```diff
@@ -240,25 +261,25 @@ status:
...
 
-1. Building AWS Policy Doc from Role definition in the spec
-2. Attempting to validate the Role in AWS by creating the Role, and deleting it if successful
-3. Setting the status to Valid or Failed
-4. If the status is Valid or Failed, stop all reconciling
-5. If an AWSFederatedRole is deleted, cleaning up any instance of the Role in AWS by cleaning up any AWSFederatedAccountAccesses using the AWSFederatedRole 
+1. Building AWS `Policy` Doc from `Role` definition in the spec.
+2. Attempting to validate the Role in AWS by creating the `Role`, and deleting it if successful.
+3. Setting the status to `Valid` or `Failed`.
+4. If the status is Valid or Failed, stop all reconciling.
```
Comment (dkeohane) [inline docs/3.0-Controllers.md:281]: `Valid` or `Failed`, like the line above, not Valid or Failed

File: docs/3.0-Controllers.md [hunk 4]
```diff
@@ -240,25 +261,25 @@ status:
...
-Updated in the accountClaim-controller
+Updated in the `AccountClaim` controller:
 
 ```txt
 MetricTotalAccountClaimCRs
 ```
 ## 3.4 AWSFederatedRole Controller
 
-The AWSFederatedRole-controller is triggered when an AWSFederatedRoke is created in any namespace. It is responsible for following behaviours:
+The `AWSFederatedRole-controller` is triggered when an `AWSFederatedRoke` is created in any namespace. It is responsible for the following behaviors:
```
Comment (dkeohane) [inline docs/3.0-Controllers.md:276]: AWSFederatedRoke != AWSFederatedRole

File: docs/3.0-Controllers.md [hunk 5]
```diff
@@ -309,22 +327,22 @@ for more information: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_po
...
+* `state` is the current state of the CR. Possible values are `Valid` and `Failed`.
 
 ### 3.4.4. Metrics
 
 None
 
 ## 3.5 AWSFederatedAccountAccess Controller
 
-The AWSFederatedAccountAccess-controller is triggered when an accountClaim is created in any namespace. It is responsible for following behaviours:
+The AWSFederatedAccountAccess-controller is triggered when an `AccountClaim` is created in any namespace. It is responsible for the following behaviors:
```
Comment (dkeohane) [inline docs/3.0-Controllers.md:339]: `AWSFederatedAccountAccess-controller`

**Review by dkeohane — COMMENTED** [review]
Awesome job :tractor: 
Mainly minor cleanup comments, but much more of a fluid read than before :)

File: docs/2.0-Custom-Resources.md [hunk 6]
```diff
@@ -88,7 +88,7 @@ spec:
 
 ## 2.4. AWSFederatedAccountAccess CR
 
-The AWSFederatedAccountAccess CR creates an instance of an AWSFederatedRole in AWS and allows the target IAM account to assume it
+The `AWSFederatedAccountAccess` CR creates an instance of an `AWSFederatedRole` in AWS and allows the target IAM account to assume its role.
```
[the remark this answers is not in the document]
Comment (dastergon) [reply docs/2.0-Custom-Resources.md:91]: just added a link in the 3rd page with the Controllers

**Comment by dastergon** [comment]
@dkeohane thanks for the suggestions! I addressed them and squashed the commit.

**Comment by rogbas** [comment]
@lisa @rogbas review please

**Pull request merged**